drivers/i2c/i2c_omap.c

```diff
@@ -0,0 +1,947 @@
+static int __maybe_unused i2c_omap_transfer_irq(const struct device *dev, struct i2c_msg msgs[],
+						uint8_t num_msgs, uint16_t addr)
+{
+	return i2c_omap_transfer_main(dev, msgs, num_msgs, false, addr);
+}
+
+/**
+ * @brief OMAP I2C transfer function using polling.
+ *
+ * This function performs the I2C transfer using the OMAP I2C controller
+ * in polling mode. It calls the common transfer function with the
+ * specified messages, number of messages, and target address.
+ *
+ * @param dev Pointer to the I2C device structure.
+ * @param msgs Array of I2C messages to be transferred.
+ * @param num_msgs Number of I2C messages in the array.
+ * @param addr Target address.
+ * @return 0 on success, negative error code on failure.
+ */
+static int __maybe_unused i2c_omap_transfer_polling(const struct device *dev, struct i2c_msg msgs[],
+						    uint8_t num_msgs, uint16_t addr)
+{
+	return i2c_omap_transfer_main(dev, msgs, num_msgs, true, addr);
+}
+
+static const struct i2c_driver_api i2c_omap_api = {
+	.transfer = i2c_omap_transfer_polling,
+	.configure = i2c_omap_configure,
+	.recover_bus = i2c_omap_recover_bus,
+};
+
+#define I2C_OMAP_INIT(inst)                                                                        \
+	LOG_INSTANCE_REGISTER(omap_i2c, inst, 4);                                                  \
+	static void i2c_omap_##inst##_init(const struct device *dev);                              \
+	static const struct i2c_omap_cfg i2c_omap_cfg_##inst = {                                   \
+		.base = DT_INST_REG_ADDR(inst),                                                    \
+		.irq = DT_INST_IRQN(inst),                                                         \
+		.init_func = i2c_omap_##inst##_init,                                               \
+		.speed = DT_INST_PROP(inst, clock_frequency),                                      \
+	};                                                                                         \
+                                                                                                   \
+	static struct i2c_omap_data i2c_omap_data_##inst;                                          \
+                                                                                                   \
+	I2C_DEVICE_DT_INST_DEFINE(inst, i2c_omap_init, NULL, &i2c_omap_data_##inst,                \
+				  &i2c_omap_cfg_##inst, POST_KERNEL, CONFIG_I2C_INIT_PRIORITY,     \
+				  &i2c_omap_api);                                                  \
+                                                                                                   \
+	static void i2c_omap_##inst##_init(const struct device *dev)                               \
+	{                                                                                          \
+		IRQ_CONNECT(DT_INST_IRQN(inst), 0, i2c_omap_isr, DEVICE_DT_INST_GET(inst), 0);     \
+                                                                                                   \
```

Comment:
you will need to honor IRQ flags here, not hardcode as 0 always.

Reply:
Updated!

Reply:
Looks like this change was missed? Also do you really need to enable the irq during init given that you only support polled mode?

Reply:
Not really,
Will drop it for now and add it with IRQ support